src/servers/grpc_server.cc

```diff
@@ -1476,40 +1476,56 @@ SharedMemoryControlHandler::Process(Handler::State* state, bool rpc_ok)
     TRTSERVER_SharedMemoryBlock* smb = nullptr;
 
     TRTSERVER_Error* err = nullptr;
-    switch (request.type()) {
-      case SharedMemoryControlRequest::REGISTER:
-        err = smb_manager_->Create(
-            &smb, request.shared_memory_region().name(),
-            request.shared_memory_region().shm_key(),
-            request.shared_memory_region().offset(),
-            request.shared_memory_region().byte_size());
-        if (err == nullptr) {
-          err = TRTSERVER_ServerRegisterSharedMemory(trtserver_.get(), smb);
-        }
-        break;
-      case SharedMemoryControlRequest::UNREGISTER:
-        err = smb_manager_->Remove(&smb, request.shared_memory_region().name());
-        if ((err == nullptr) && (smb != nullptr)) {
-          err = TRTSERVER_ServerUnregisterSharedMemory(trtserver_.get(), smb);
-          TRTSERVER_Error* del_err = TRTSERVER_SharedMemoryBlockDelete(smb);
-          if (del_err != nullptr) {
-            LOG_ERROR << "failed to delete shared memory block: "
-                      << TRTSERVER_ErrorMessage(del_err);
-            TRTSERVER_ErrorDelete(del_err);
-          }
+    if (request.has_register_()) {
```

Comment:
has_register_? Does that function even exist?

Reply:
register gets named as register_ due to a keyword issue or conflict.